### PR TITLE
Fix aa being upside down on swapped y-major slopes

### PR DESCRIPTION
**Pull request by Jaklyy**
fixes kyurem's face on the title screen, also fixes crossed quads.
image
fixed
image
console screenshot
image

### CODE REVIEW

**Comment by Jaklyy** [comment]
draft pr because it needs more testing, specifically i'd like more than two examples of this bug occurring, i think it's good though? also not sure whether i wanna do a light restructuring of some of the relevant code or if that's just unnecessary to worry about.

**Comment by Jaklyy** [comment]
added improved handling for swapped vertical and x-major slopes.
should all be console accurate (or at least closer in the case of x-major slopes)

File: src/GPU3D_Soft.cpp [hunk 1]
```diff
@@ -753,8 +753,8 @@ void SoftRenderer::RenderShadowMaskScanline(RendererPolygon* rp, s32 y)
         interp_start = &rp->SlopeR.Interp;
         interp_end = &rp->SlopeL.Interp;
 
-        rp->SlopeR.EdgeParams_YMajor(&l_edgelen, &l_edgecov);
-        rp->SlopeL.EdgeParams_YMajor(&r_edgelen, &r_edgecov);
+        rp->SlopeR.EdgeParams(&l_edgelen, &l_edgecov, 1);
```
Comment (RSDuck) [inline src/GPU3D_Soft.cpp:756]: please use true and false

**Comment by Jaklyy** [comment]
took the time to rewrite it slightly to hopefully force it to pre-compile the if statements, wasn't sure if they would on its own the way i had done it. feel free to tell me if there's a better way to do this/if you think this is worse from a readability/optimization standpoint

File: src/GPU3D_Soft.cpp (hunk 1 above)
Comment (Jaklyy) [reply src/GPU3D_Soft.cpp]: done

**Pull request merged**